pycat.py

```diff
@@ -212,8 +212,13 @@ def run(self):
                         self.handle_from_telnet()
                     elif fd == self.socketToPipeR:
                         self.handle_from_pipe()
-        except Exception as e:
-            traceback.print_exc()
+        except (Exception, KeyboardInterrupt) as error:
+            if type(error) is EOFError:
+                pass  # This simply means the telnet connection was closed.
```

Comment:
Does it now just hang when connection is closed?

Reply:
Yup. You have to C^c again and that crashes somewhere in the threading module. I can't seem to hunt that one down.

Reply:
I think so. It makes the "real" crash stand out, and this code highlights the place to put a raise once that's fixed. I do think I should add a message though, based on your feedback. I'll send another commit in about an hour

Reply:
I've added a message for clarity.  I still can't hunt down the reason for threads crashing on interrupt.  It's odd that I can't seem to catch that KeyboardInterrupt, and it's odd that it's written with a ':' after it.